pkg/bgp/speaker/speaker.go

```diff
@@ -121,6 +142,9 @@ func (s *Speaker) OnUpdateService(svc *slim_corev1.Service) {
 	s.services[svcID] = svc
 	s.Unlock()
 
+	if s.shutDown() {
+		return
+	}
```

Comment:
I thought that I refactored the other subscriber implementations to return errors as well, but I'm only seeing the node subscriber methods returning errors? If the others haven't been implemented, I think it's worth doing a refactor commit to do that. Doesn't have to be part of this specific PR though.

Reply:
I'll do this in this PR.